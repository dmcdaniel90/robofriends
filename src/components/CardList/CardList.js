import React from "react";
import Card from "../Card/Card";

const CardList = ({ robots }) => {
  return (
    <div data-testid="card-list">
      {
        robots.map((user, i) => {
          return (
            <Card
              id={robots[i].id}
              name={robots[i].name}
              email={robots[i].email}
              key={i}
            />
           
          );
        })
      }
    </div>
  )
}

export default CardList;