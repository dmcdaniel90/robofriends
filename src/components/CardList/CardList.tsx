import Card from "../Card/Card";
import { Robot } from "../../types/common";

type CardListProps = {
  robots: Array<Robot>
}

const CardList = ({ robots }: CardListProps) => {
  return (
    <div data-testid="card-list">
      {
        robots.map((user: Robot, i: number) => {
          return (
            <Card
              id={robots[i].id}
              name={robots[i].name}
              email={robots[i].email}
              key={i}
            />
           
          );
        })
      }
    </div>
  )
}

export default CardList;