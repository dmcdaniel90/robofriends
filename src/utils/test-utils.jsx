import { createRoot } from 'react-dom/client';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { thunk } from 'redux-thunk';
import { searchRobots, requestRobots } from '../reducers';

const rootReducer = combineReducers({ searchRobots, requestRobots })

const AllProviders = ({ children }) => {
  return (
    <Provider store={createStore(rootReducer, applyMiddleware(thunk))}>
      {children}
    </Provider>
  );
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';

export { customRender as render };


