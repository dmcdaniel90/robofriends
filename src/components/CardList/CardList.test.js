import { render } from '../../utils/test-utils';
import CardList from './CardList';

describe('CardList', () => {
  it('expect to render CardList component', () => {
    const mockRobots = [ //This is needed because CardList expects a robots prop that is mapped over. Testing cannot be done for mapped components without a mock prop.
      {
        id: 1,
        name: 'John Snow',
        username: 'JohnJohn',
        email: 'john@gmail.com'
      }
    ]
    
    const {container} = render(<CardList robots={mockRobots} />)
    expect(container.firstChild).toMatchSnapshot();
  })
})