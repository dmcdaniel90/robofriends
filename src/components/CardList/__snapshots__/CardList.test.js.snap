// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList expect to render CardList component 1`] = `
<div
  data-testid="card-list"
>
  <div
    class="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
  >
    <img
      alt="robots"
      data-testid="card"
      height="200px"
      src="placeholder.jpg"
      width="200px"
    />
    <div>
      <h2>
        John Snow
      </h2>
      <p>
        john@gmail.com
      </p>
    </div>
  </div>
</div>
`;
